apps/functions/deadline-submissions/deadline-submission-command/index.js

```diff
@@ -0,0 +1,46 @@
+import { BlobStorageClient } from '@pins/blob-storage-client';
+import config from './config';
+import api from './back-office-api-client';
+
+const { storageUrl, submissionsContainer, uploadsContainer } = config;
+
+/**
+ *
+ * @param {import('@azure/functions').Context} context
+ * @param {import('@pins/api/src/message-schemas/commands/new-deadline-submission').DeadlineSubmission} msg
+ */
+export default async function (context, msg) {
+	context.log('Handle new deadline submission', msg);
+
+	const client = BlobStorageClient.fromUrl(storageUrl);
+
+	const destinationBlobName = `${msg.deadline}/${msg.submissionType}/${msg.blobGuid}/${msg.documentName}`;
+
+	const result = await client.copyFile({
+		sourceContainerName: submissionsContainer,
+		sourceBlobName: `${msg.blobGuid}/${msg.documentName}`,
+		destinationContainerName: uploadsContainer,
+		destinationBlobName
+	});
+
+	if (result !== 'success') {
+		// TODO: Publish failure message to service bus
+		context.log('Copying blob failed', result);
+	}
+
+	const properties = await client.getBlobProperties(uploadsContainer, destinationBlobName);
+
+	const caseID = await api.getCaseID(msg.caseReference);
+	const folderID = await api.getFolderID(caseID, msg.deadline, msg.submissionType);
+
+	await api.submitDocument({
```

Comment:
Currently it looks like code in the web repo uploads documents to blob storage directly, and then calls the API with a reference to the added blob. So yeah it takes 2 calls (blob storage + documents API) otherwise you end up with an orphaned blob document.

Reply:
I think it's the other way around right? The web repo calls the endpoint to get a URL, the endpoint creates a document and then returns the URL to upload against, and then the client (currently the web app, but in this case your function) uploads against that URL.

Reply:
I've changed the order and now it copies the source blob to the destination given in the API response.